packages/kit/test/apps/basics/test/client.test.js

```diff
@@ -905,8 +905,10 @@ test('+server.js next to +page.svelte works', async ({ page }) => {
 	expect(await page.textContent('p')).toBe('Hi');
 
 	for (const method of ['GET', 'PUT', 'PATCH', 'POST', 'DELETE']) {
-		page.click(`button:has-text("${method}")`);
-		await page.waitForResponse('/routing/endpoint-next-to-page');
-		expect(await page.textContent('pre')).toBe(method);
+		await page.click(`button:has-text("${method}")`);
+		await page.waitForFunction(
+			(method) => document.querySelector('pre').textContent === method,
+			method
+		);
```

Comment:
I'm surprised this reduces flakiness when prior to https://github.com/sveltejs/kit/pull/6947 it was using this, which should in theory be equivalent:

```suggestion
		await expect(page.locator('pre')).toHaveText(method);
```

Reply:
unbelievable — making that change reintroduced flakiness. the hell. https://github.com/sveltejs/kit/actions/runs/3107914512/jobs/5036541318#step:8:397

Reply:
Yeah it's super strange, this is basically the way to un-flake it. Playwright bug..?

Reply:
maybe we need to set the `polling` option? https://playwright.dev/docs/api/class-page#page-wait-for-function

otherwise it will wait until there's a `requestAnimationFrame`. I'm not that familiar with `requestAnimationFrame`. Is it something that will happen periodically in the browser without animations running or do we need to fake a tick?
```
export function tick(/** @type {import('playwright-core').Page} */ page) {
	return page.evaluate(() => new Promise(requestAnimationFrame));
}
```

Reply:
This is wild. The page actually says "GET", but playwright is getting "loading" eventhough it's not on the page. This really does feel like a Playwright bug. I guess I'll try to see if I can reproduce it in a simpler project

![Screenshot from 2022-09-23 08-14-46](https://user-images.githubusercontent.com/322311/191995405-fa16d87b-6efc-45fa-bdcc-d930dc3c170f.png)